### DISCUSSION TITLE
Question: Windows Forms Designer default font

### DISCUSSION THREAD

**Discussion by coderb** [dotnet/winforms#7010] [Q&A]
### Environment

vs2022

### .NET version

net472 & net6.0-windows

### Did this work in a previous version of Visual Studio and/or previous .NET release?

n/a

### Issue description

I have Form classes that were designed under net472 which had a default font of `new Font(FontFamily.GenericSansSerif, 8.25f)`.

When compiling/running under net6 I call `Application.SetDefaultFont(new Font(FontFamily.GenericSansSerif, 8.25f))` at the beginning of my program.

If I set my project `TargetFramework` to net472, the windows forms editor works as expected.

If I set my project `TargetFramework` to `net6.0-windows`, the windows forms editor wants to change
`this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);` -> `this.AutoScaleDimensions = new System.Drawing.SizeF(7F, 15F);`
along with changing every control's `Location` and `Size` and adding lines to set `.Margin`.

Is there a way to configure the designer to make it aware that my application uses an alternative default font under net6 so it doesn't make tons of changes to existing files when they are edited?

### Steps to reproduce

see description

**Comment by RussKie** [comment]
You need to use `ApplicationConfiguration.Initialize()` API.
Please refer to https://aka.ms/applicationconfiguration for details.

**Reply by LesmesWasNotHere** [reply]
I am facing the same problem, but in a very large Visual Basic .Net application that we are trying to migrate from Framework 4.7.2 to .Net 6.

In C# applications, ApplicationConfiguration.Initialize() is generated at compile time and works ok, but in VB.Net that class is not generated and the suggested solution does not work.

Is there any way to get ApplicationConfiguration generated in VB.Net applications?

**Reply by kev160967** [reply]
I'm having the same issue. Very large winforms app with the UI layer written in VB.Net. On migrating to Net6 all our forms have been rescaled. This is particularly problematic as we remember users form positions and sizes, so many forms are now opening to small to display their contents. I put the suggested change in the project files but it made no difference to the designer. After seeing this comment I created a test solution with both a C# and a VB project and applied the change to both project files. In the VB project the designer still used Segoe 9pt, whilst the C# project uses MS Sans Serif 8.25.

Are there any suggestions to resolve this? If I use code to set the font at runtime it makes it extremely difficult to design forms because you can no longer rely on the designer to match what the user will experience. I also can't individually go into each form and set a specific font (and what a tedious process that would be) because the designer will already have rescaled the form when it loaded it, and the damage is already done.

**Reply by RussKie** [reply]
To give you heads up, a change that enables a _limited_ support of [`ApplicationDefaultFont`](https://learn.microsoft.com/dotnet/core/project-sdk/msbuild-props-desktop#applicationdefaultfont) for VB app in design-time scenarios is expected to become available in VS 2022 v17.7 band. I expect that the team will provide more information closer to the release.

**Comment by coderb** [comment]
@RussKie thanks, that was helpful. i added the settings into my project file for default font, etc and the `Location,Size,AutoScaleDimensions` changes are no longer happening when using the net6 designer.

I am however still getting calls to `this.mycontrol.Margin = new System.Windows.Forms.Padding(4, 3, 4, 3);` for some controls though :-(. 

From what I can tell this only happens on controls that subclass `UserControl`.
Thanks!

-----
edit:
it looks like there is some non-trivial logic happening with `Margin` defaults in `Control.ctor()`

not sure what's up with this... do i need to do something else to not get all the excess `.Margin =` lines in my designer code?

```
            // We baked the "default default" margin and min size into CommonProperties
            // so that in the common case the PropertyStore would be empty.  If, however,
            // someone overrides these Default* methads, we need to write the default
            // value into the PropertyStore in the ctor.

            if (DefaultMargin != CommonProperties.DefaultMargin)
            {
                Margin = DefaultMargin;
            }

        protected virtual Padding DefaultMargin => CommonProperties.DefaultMargin;

```

**Reply by RussKie** [reply]
@dreddy-work is currently working through various dpi/scaling-related issues. If you can create a small targeted sample for your issue - please open a new issue, and we'll look into it.

**Comment by coderb** [comment]
The issue with `.Margin` is feeling like a bug to me after further investigation (unless there is some magical setting I am missing).


Using the net472 forms designer to edit a `Form` (created in net472) does not change the child control `.Margin` from the default of `new Padding(3)`.  I also checked at runtime after `OnLoad` is called and the `Margin` remained at the default of `new Padding(3)`.

Switching to the net6 forms designer to edit the `Form` changes the child controls to ad calls to `.Margin = new System.Windows.Forms.Padding(4, 3, 4, 3);` inside `InitializeComponent`.

Without editing the `Form` but doing a runtime inspection of `.Margin` during `OnLoad` shows the correct value of `new Padding(3)`.

I don't know where the 4/3/4/3 padding is being introduced.  Seems like a designer bug.

**Reply by kirsan31** [reply]
> The issue with `.Margin` is feeling like a bug to me after further investigation (unless there is some magical setting I am missing).

I think that in order to get some response, it's better to open an issue...
